mlflow/pytorch/_lightning_autolog.py

```diff
@@ -287,6 +291,136 @@ def on_test_end(self, trainer, pl_module):
         self.metrics_logger.flush()
 
 
+_LATEST_CHECKPOINT_ARTIFACT_TAG_KEY = "_latest_checkpoint_artifact"
+
+
+class __MLflowModelCheckpointCallback(pl.Callback, metaclass=ExceptionSafeAbstractClass):
+
+    def __init__(
+        self,
+        client,
+        run_id,
+        monitor,
+        mode,
+        save_best_only,
+        save_weights_only,
+        save_freq,
+    ):
+        self.client = client
+        self.run_id = run_id
+        self.monitor = monitor
+        self.mode = mode
+        self.save_best_only = save_best_only
+        self.save_weights_only = save_weights_only
+        self.save_freq = save_freq
+        self.latest_checkpoint_timestamp = time.time()
+        self.last_monitor_value = None
+
+    def _is_new_checkpoint_better(self, new_monitor_value):
+        if self.last_monitor_value is None:
+            return True
+
+        if self.mode == "min":
+            return new_monitor_value <= self.last_monitor_value
+
+        if self.mode == "max":
+            return new_monitor_value >= self.last_monitor_value
+
+        assert False, "Illegal __MLflowModelCheckpoint config."
+
+    def _save_checkpoint_rank_zero_only(self, trainer: "pl.Trainer", filepath: str):
+        checkpoint = trainer._checkpoint_connector.dump_checkpoint(self.save_weights_only)
+        trainer.strategy.save_checkpoint(checkpoint, filepath)
+
+    def _check_and_save_checkpoint_if_needed(self, trainer: "pl.Trainer"):
+        current_epoch = trainer.current_epoch
+        metric_dict = {k: float(v) for k, v in trainer.callback_metrics.items()}
+
+        if self.save_best_only:
+            if self.monitor not in metric_dict:
+                # "save-best-only" requires comparing the monitor metric value,
```

Comment:
I prefer to use:
```
                _logger.error(
                    "If MLflowModelCheckpoint 'save_best_only' config is True, it requires to "
                    "compare the monitored metric value, but the provided monitored metric value "
                    "is not available."
                )
```
instead of raise exception,

because the default checkpoint configs are:
```
    checkpoint=True,
    checkpoint_monitor="val_loss",
    checkpoint_mode="min",
    checkpoint_save_best_only=True,
```
but sometimes the model might not log "val_loss" metric, so that if it raise exception here, it breaks the whole autologging , but we don't need to break other part autologging (e.g. log params / metrics)

Reply:
makes sense to me! 

On top of that, may we call out "checkpoint logging is skipped" in the error message?